cmake/Makefile

```diff
@@ -0,0 +1,349 @@
+# CMAKE generated file: DO NOT EDIT!
```

Comment:
Are generated files typically included in source control? Or should we expect developers to run cmake on their own machines to generate these?